### PR TITLE
Add Organization functionality feature added

### PR DESCRIPTION
**Pull request by poojakarma**
Steps for Implementation

Step 1: Add new button for Add organization
Step 2: Create new child component for add ORG inside parent SB-ORG
Step 3: Call this Add component from parent component
Step 4: Design UI for ADD ORG
Step 5: Apply validation on form field
Step 6: create service to Call add ORG API
Step 7: create function in component in ADD ORG component
Step 8: Submit data using API
Step 9 : newly added user will be visible in the table (by using observable)
Step 10 : Newly added data will visible on top
Step 11: Data will show in table from newly to oldest submitted

Video: https://drive.google.com/file/d/1Gq6aLMPJsRawh13WF3YeP53cF3mpTVP-/view?usp=share_link

### CODE REVIEW

File: src/app/sb-admin/components/pages/sb-organization/add-or-edit-org/add-or-edit-org.component.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      description: ['', Validators.required],
+      channel: ['', Validators.required],
+      organisationType: ['', Validators.required],
+      isRootOrg: true,
+      isTenant: true
+    })
+  }
+  ngOnInit() {
+    this.organizationType = [
+      { name: 'school', value: 'school' },
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-or-edit-org/add-or-edit-org.component.ts]: why this is hardcoded here

File: src/app/sb-admin/components/pages/sb-organization/add-or-edit-org/add-or-edit-org.component.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  cancel() {
+    this.ref.close();
+  }
+
+  saveOrg() {
+    this.submitted = true;
+    if (this.addEditOrgForm.invalid) {
+      this.messages = [
+        { severity: 'error', detail: 'Fields should not be blank' }
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-or-edit-org/add-or-edit-org.component.ts]: use language constants

File: src/app/sb-admin/components/pages/sb-organization/add-or-edit-org/add-or-edit-org.component.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    this.addOrgservice.addOrg(body).subscribe((response) => {
+      const id = response.result.organisationId
+      const updatedFormValues = {
+        ...this.addEditOrgForm.value,
+        id: id
+      };
+      this.ref.close(updatedFormValues);
+    }, (error: any) => {
+      if (error.status = 400) {
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-or-edit-org/add-or-edit-org.component.ts]: are you assigning the 400 value to error.status?

File: src/app/sb-admin/components/pages/sb-organization/add-or-edit-org/add-or-edit-org.component.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        this.messages = [
+          { severity: 'error', detail: 'Channel Already Exit' }
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-or-edit-org/add-or-edit-org.component.ts]: use language constants

File: src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts [hunk 5]
```diff
@@ -11,39 +14,59 @@ import { Subscription } from 'rxjs';
...
+      },
+      (error: any) => {
+        console.log(error);
+        this.loading = false;
+      }
     );
   }
 
+  addOrg() {
+    this.ref = this.dialogService.open(AddOrEditOrgComponent, {
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts]: move this object in variable declaration section

File: src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts [hunk 6]
```diff
@@ -11,39 +14,59 @@ import { Subscription } from 'rxjs';
...
+    });
+    this.ref.onClose.subscribe((newOrganizationData: any) => {
+      if (newOrganizationData) {
+        this.organizationDetail.unshift(newOrganizationData);
+        this.displaySuccessMessage('Organization added successfully');
+      }
+    });
+  }
+  displaySuccessMessage(message: string) {
+    this.messageService.add({ severity: 'success', summary: 'Success', detail: message });
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts]: what message you are showing here?

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts [hunk 7]
```diff
@@ -11,39 +15,55 @@ import { Subscription } from 'rxjs';
...
-      this.organizationDetail = data.result.response.content;
-      this.loading = false;
-    },
-    (error:any)=>{
-      console.log(error);
-      this.loading = false;
-    }
+        this.organizationDetail = data.result.response.content;
+        this.organizationDetail.sort((startDate:any ,endDate :any)=>
+          new Date(endDate.createdDate).getTime() - new Date(startDate.createdDate).getTime());
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts:48]: what is the use of this? you should pass the filter from request body while calling the API

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts (hunk 7 above)
Comment (poojakarma) [reply src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts:48]: actually sir API filter not working that why i used that code to sort

Comment (gouravmore) [reply src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts:48]: @nidhikachhawa24 Please confirm.

Comment (nidhikachhawa24) [reply src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts:48]: @gouravmore Yes the sorting in API is not working. I've checked myself on this and different API's also. That's why suggested to add sorting in the code itself.

Comment (gouravmore) [reply src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts:48]: ok

**Review by gouravmore — APPROVED** [review]

**Review by gouravmore — APPROVED** [review]

**Pull request merged**